src/MagnumPlugins/AssimpImporter/Test/Test.cpp

```diff
@@ -0,0 +1,257 @@
+        {1.0f, 0.15f, 0.45f}};
+
+    for (int i : {0, 1, 2}) {
+        std::optional<Trade::LightData> light = importer.light(i);
+        CORRADE_VERIFY(light);
+        CORRADE_COMPARE(light->type(), types[i]);
+        CORRADE_COMPARE(light->color(), colors[i]);
+        CORRADE_COMPARE(light->intensity(), 1.0f);
+
+
+        std::unique_ptr<Trade::ObjectData3D> lightObject = importer.object3D(i + 1);
+        CORRADE_COMPARE(lightObject->instanceType(), ObjectInstanceType3D::Light);
+        CORRADE_COMPARE(lightObject->instance(), i);
+    }
+
+    if(aiGetVersionMajor() <= 3 && aiGetVersionMinor() < 1) {
+        /* Cannot load lights with undefined light type yet */
+        return;
+    }
+
+    std::ostringstream out;
+    Error redirectError{&out};
+
+    CORRADE_VERIFY(!importer.light(3));
+    CORRADE_COMPARE(out.str(), "Trade::AssimpImporter::light(): Undefined light type is not supported.\n");
+}
+
+void AssimpImporterTest::material() {
+    AssimpImporter importer;
+    CORRADE_VERIFY(importer.openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR, "mesh_material.dae")));
+
+    CORRADE_COMPARE(importer.materialCount(), 1);
+}
+
+void AssimpImporterTest::mesh() {
+    AssimpImporter importer;
+    CORRADE_VERIFY(importer.openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR, "mesh.dae")));
+
+    CORRADE_COMPARE(importer.mesh3DCount(), 1);
+
+    std::optional<Trade::MeshData3D> mesh = importer.mesh3D(0);
+    CORRADE_VERIFY(mesh);
+    CORRADE_VERIFY(mesh->isIndexed());
+    CORRADE_COMPARE(mesh->primitive(), MeshPrimitive::Triangles);
+    CORRADE_COMPARE(mesh->positionArrayCount(), 1);
+    CORRADE_COMPARE(mesh->normalArrayCount(), 1);
+    CORRADE_COMPARE(mesh->textureCoords2DArrayCount(), 0);
+    CORRADE_COMPARE(mesh->colorArrayCount(), 0);
+
+    CORRADE_COMPARE(mesh->positions(0), (std::vector<Vector3>{
+            {-1.0f, 1.0f, 1.0f}, {-1.0f, -1.0f, 1.0f}, {1.0f, -1.0f, 1.0f}}));
+    CORRADE_COMPARE(mesh->indices(), (std::vector<UnsignedInt>{0, 1, 2}));
+
+    std::unique_ptr<Trade::ObjectData3D> meshObject = importer.object3D(1);
+    CORRADE_COMPARE(meshObject->instanceType(), ObjectInstanceType3D::Mesh);
+    CORRADE_COMPARE(meshObject->instance(), 0);
+}
+
+void AssimpImporterTest::scene() {
+    AssimpImporter importer;
+    CORRADE_VERIFY(importer.openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR, "scene.dae")));
+
+    CORRADE_COMPARE(importer.sceneCount(), 1);
+    CORRADE_COMPARE(importer.defaultScene(), 0);
+    std::optional<Trade::SceneData> data = importer.scene(0);
+    CORRADE_VERIFY(data);
+
+    CORRADE_COMPARE(data->children2D(), {});
+    CORRADE_COMPARE(data->children3D(), {0});
+
+    std::unique_ptr<Trade::ObjectData3D> rootObject = importer.object3D(0);
+    CORRADE_COMPARE(rootObject->children(), {1});
+    //CORRADE_COMPARE(rootObject->instanceType(), ObjectInstanceType3D::Mesh); // Should be Empty, but Assimp creates mesh for root node for some reason
```

Comment:
I think the rootObject has a mesh, but I don't know why (aiScene->mRoot has one already)